crates/bevy_ecs/src/query/fetch.rs

```diff
@@ -444,6 +448,49 @@ pub trait WorldQueryGats<'world> {
     type Fetch;
 }
 
+/// An extension of [`WorldQuery`] for batched queries.
+pub trait WorldQueryBatch<const N: usize, const ALIGN: usize>:
+    for<'w> WorldQueryBatchGats<'w, N, ALIGN> + for<'w> WorldQueryGats<'w>
+{
+    /// Retrieve a batch of size `N` with desired alignment `ALIGN` from the current table.
+    /// # Safety
+    ///
+    /// `table_row_start` is a valid table row index for the current table
+    /// `table_row_start` + `N` is a valid table row index for the current table
+    /// `table_row_start` is a multiple of `N`
+    ///
+    /// Must always be called _after_ [`WorldQuery::set_table`].
+    unsafe fn table_fetch_batched_aligned<'w>(
+        fetch: &mut <Self as WorldQueryGats<'w>>::Fetch,
+        table_row_start: usize,
+        len: usize,
+    ) -> <Self as WorldQueryBatchGats<'w, N, ALIGN>>::FullBatch;
+}
+
+impl<const N: usize, const ALIGN: usize> WorldQueryBatch<N, ALIGN> for Entity
```

Comment:
this impl should be with the other worldquery impls for `Entity`, not in the middle of a bunch of trait and type alias definitions

Reply:
Yes, totally agree, I will see about organizing this better and possibly moving it to a new file for clarity.  Thank you!

Reply:
@BoxyUwU I've improved this a bit by reorganizing `fetch.rs` to always have the definitions in the same order for each type of query.  However, I might move these definitions to a new file with the generalized const expressions changes coming up.

Reply:
Marking this resolved as the GAT refactor cleaned all of this right up.  Still might move things to a new file for clarity.